src/emulator/storage/apis/gcloud.ts

```diff
@@ -90,9 +90,8 @@ export function createCloudEndpoints(emulator: StorageEmulator): Router {
     res.json(listResult);
   });
 
-  gcloudStorageAPI.delete("/b/:bucketId/o/:object", (req, res) => {
-    const decodedObjectId = decodeURIComponent(req.params.objectId);
```

Comment:
Do we need to decode? I think the best way to find out is to add a test with a file under a nested path (e.g. `a/b/c.txt`) and delete that

Reply:
+1 I'm pretty sure decoding is needed

Reply:
Decoding was one of the reason why it didn't work.
If you look at the other route handlers (get and patch for instance), they don't apply decoding.

Reply:
Can you add a test of deleting at a nested path so we know the behavior is verified?

Reply:
I actually rewrote the whole test to make sure to use a nested path and that the file exists and is then properly deleted.
I was also foolishly testing if the file was deleted on `fs` instead of on the bucket. This is now checked properly.
Let me know what you think.

Reply:
looks great! I think we'll be good to merge unless Yuchen has any issues

Reply:
@abeisgoat @yuchenshi
Just updated the branch to resolve the conflicts after my other PR was merged.

Reply:
Good to go.